src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -214,6 +214,42 @@ public void RunWorkerAsyncTwice()
         }
 
         [Fact]
+        public void TestCancelInsideDoWork()
+        {
+            var orignal = SynchronizationContext.Current;
```

Comment:
Did you intend to then do SynchronizationContext.SetSynchronizationContext(null)?  Otherwise, why are we capturing and restoring the context if we're not changing it?

Nit: orignal => original